cpu/fe310/cpu.c

```diff
@@ -17,379 +17,93 @@
+        SPI_INSN_CMD_EN         |          /* Enable memory-mapped flash                    */
+        SPI_INSN_ADDR_LEN(3)    |          /* 25LP03D read commands have 3 address bytes    */
+        SPI_INSN_PAD_CNT(4)     |          /* 25LP03D Table 6.11 Read Dummy Cycles = 4      */
+        SPI_INSN_CMD_PROTO(SPI_PROTO_S) |  /* 25LP03D Table 8.1 "Instruction                */
+        SPI_INSN_ADDR_PROTO(SPI_PROTO_D) | /*  Set" shows mode for cmd, addr, and           */
+        SPI_INSN_DATA_PROTO(SPI_PROTO_D) | /*  data protocol for given instruction          */
+        SPI_INSN_CMD_CODE(0xBB) |          /* Set the instruction to "Fast Read Dual I/O"   */
+        SPI_INSN_PAD_CODE(0x00);           /* Dummy cycle sends 0 value bits                */
+    /* end{code-style-ignore} */
+
+    /*
+     * The relationship between the input clock and SCK is given
+     * by the following formula (Fin is processor/tile-link clock):
+     *    Fsck = Fin/(2(div + 1))
+     */
+    uint32_t freq = cpu_freq();
+    uint32_t sckdiv = (freq - 1) / (MAX_FLASH_FREQ * 2);
+    if (sckdiv > SCKDIV_SAFE) {
+        SPI0_REG(SPI_REG_SCKDIV) = sckdiv;
     }
-
-    printf("printing the current stack of thread %" PRIkernel_pid "\n",
-           thread_getpid());
-
-#ifdef DEVELHELP
-    printf("thread name: %s\n", sched_active_thread->name);
-    printf("stack start: 0x%08x\n", (unsigned int)(sched_active_thread->stack_start));
-    printf("stack end  : 0x%08x\n", (unsigned int)(sched_active_thread->stack_start + sched_active_thread->stack_size));
-#endif
-
-    printf("  address:      data:\n");
-
-    do {
-        printf("  0x%08x:   0x%08x\n", (unsigned int) sp, (unsigned int) *sp);
-        sp++;
-        count++;
-    } while (*sp != STACK_MARKER);
-
-    printf("current stack size: %i words\n", count);
-}
-
-int thread_isr_stack_usage(void)
-{
-    return 0;
-}
-
-void *thread_isr_stack_pointer(void)
-{
-    return NULL;
-}
-
-void *thread_isr_stack_start(void)
-{
-    return NULL;
 }
 
 /**
- * @brief Call context switching at thread exit
- *
- * This is called is two situations: 1) after the initial main and idle threads
- * have been created and 2) when a thread exits.
- *
+ * @brief Initialize the CPU, set IRQ priorities, clocks, peripheral
  */
-void cpu_switch_context_exit(void)
+void cpu_init(void)
 {
-    /* enable interrupts */
-    irq_enable();
-
-    /* force a context switch to another thread */
-    thread_yield_higher();
-    UNREACHABLE();
-}
+    /* Initialize clock */
+    clock_init();
 
-void thread_yield_higher(void)
-{
-    /* Use SW intr to schedule context switch */
-    CLINT_REG(CLINT_MSIP) = 1;
+#if USE_CLOCK_HFXOSC_PLL || USE_CLOCK_HFROSC_PLL
+    /* Initialize flash memory, only when using the PLL: in this
+       case the CPU core clock can be configured to be so fast that the SPI
+       flash frequency needs to be adjusted accordingly. */
+    flash_init();
+#endif
```

Comment:
Removing this for `USE_CLOCK_HFXOSC_PLL` fixed the issue for me.

Reply:
Removed